### PR TITLE
Fix violations of a handful of Error Prone checks

### PR DESCRIPTION
**Pull request by basil**
Fixes violations of the following Error Prone checks:

- [`UseCorrectAssertInTests`](https://errorprone.info/bugpattern/UseCorrectAssertInTests)
- [`IdentityHashMapUsage`](https://errorprone.info/bugpattern/IdentityHashMapUsage)
- [`InputStreamSlowMultibyteRead`](https://errorprone.info/bugpattern/InputStreamSlowMultibyteRead)
- [`ThreeLetterTimeZoneID`](https://errorprone.info/bugpattern/ThreeLetterTimeZoneID)
- [`TypeParameterShadowing`](https://errorprone.info/bugpattern/TypeParameterShadowing)
- [`UnescapedEntity`](https://errorprone.info/bugpattern/UnescapedEntity)
- [`NonAtomicVolatileUpdate`](https://errorprone.info/bugpattern/NonAtomicVolatileUpdate)
- [`EmptyBlockTag`](https://errorprone.info/bugpattern/EmptyBlockTag)
- [`DoubleBraceInitialization`](https://errorprone.info/bugpattern/DoubleBraceInitialization)
- [`SameNameButDifferent`](https://errorprone.info/bugpattern/SameNameButDifferent)
- [`UnnecessaryParentheses`](https://errorprone.info/bugpattern/UnnecessaryParentheses)

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/ExtensionList.java
Comment (daniel-beck) [inline core/src/main/java/hudson/ExtensionList.java:251]:
```suggestion
    private <X> boolean removeComponent(Collection<ExtensionComponent<X>> collection, Object x) {
        for (ExtensionComponent<X> c : collection) {
            if (c.getInstance().equals(x)) {
```

I would expect this to be called `t` because it's "a `T`" (or otherwise `T` related), so should be changed accordingly (or to something else entirely)

File: core/src/main/java/hudson/model/User.java
Comment (daniel-beck) [inline core/src/main/java/hudson/model/User.java:871]: IMO parens around pre/post increment seem useful for readability.

File: core/src/main/java/hudson/util/UnbufferedBase64InputStream.java
Comment (daniel-beck) [inline core/src/main/java/hudson/util/UnbufferedBase64InputStream.java:48]:
```suggestion
        return decoded[pos++] & 0xFF;
```

File: test/src/test/java/hudson/console/ConsoleAnnotatorTest.java
Comment (daniel-beck) [inline test/src/test/java/hudson/console/ConsoleAnnotatorTest.java:222]: Really? More readable? At least do 
```suggestion
                text.addMarkup(0, 5, "<b tag=" + n++ + ">", "</b>");
```

(But I still prefer to keep parens around these)

File: core/src/main/java/hudson/util/SecretRewriter.java
Comment (daniel-beck) [inline core/src/main/java/hudson/util/SecretRewriter.java:159]:
```suggestion
                    if (count++ % 100 == 0)
```

File: core/src/main/java/hudson/ExtensionList.java
Comment (basil) [reply core/src/main/java/hudson/ExtensionList.java]: Better yet, this doesn't need to be a generic method in the first place, because the only values being passed in by callers are already of type `T`. So we can keep using `T` and just remove the type parameter from the method.

File: test/src/test/java/hudson/console/ConsoleAnnotatorTest.java
Comment (basil) [reply test/src/test/java/hudson/console/ConsoleAnnotatorTest.java]: I think this is more a reflection of Kohsuke's unique coding style than anything else…

File: core/src/main/java/hudson/model/User.java
Comment (basil) [reply core/src/main/java/hudson/model/User.java:871]: As long as there are spaces around each + sign this doesn't seem too bad, and keeping the linter happy has other benefits like preventing further violations from being introduced. (I'm not introducing any enforcement in this PR, but I might try enabling Checkstyle's [`UnnecessaryParentheses`](https://checkstyle.sourceforge.io/config_coding.html#UnnecessaryParentheses) check in a future PR.)

**Comment by timja** [comment]
conflicts

**Review by timja — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
